assets/src/components/vehicleIcon.tsx

```diff
@@ -1,16 +1,299 @@
 import React from "react"
 
-const VehicleIcon = ({ scale }: { scale?: number }) => {
-  const scaleAmount = scale || 1.0
+export enum Orientation {
+  Up,
+  Down,
+  Left,
+  Right,
+}
+
+export enum Size {
+  Small,
+  Medium,
+  Large,
+}
+
+export interface Props {
+  size: Size
+  orientation: Orientation
+  label?: string
+  variant?: string | null
+}
+
+/*
+Size of the triangle for calculation of the viewbox and positioning the label
+Sizes are before scaling, relative to the center of the triangle
+*/
+const X_CENTER_TO_SIDE = 22
+const Y_CENTER_TO_POINT = 16
+const Y_CENTER_TO_BASE = 20
+
+export const VehicleIcon = ({ size, orientation, label, variant }: Props) => {
+  // shrink the viewbox to fit around the triangle and label
+  const scale = scaleForSize(size)
+  const labelBgWidth = label ? labelBackgroundWidth(size) : 0
+  const labelBgHeight = label ? labelBackgroundHeight(size) : 0
+  let left = 0
+  let right = 0
+  let top = 0
+  let bottom = 0
+  switch (orientation) {
+    case Orientation.Up:
+      left = -scale * X_CENTER_TO_SIDE
+      top = -scale * Y_CENTER_TO_POINT
+      right = scale * X_CENTER_TO_SIDE
+      bottom = scale * Y_CENTER_TO_BASE + labelBgHeight
+      break
+    case Orientation.Down:
+      left = -scale * X_CENTER_TO_SIDE
+      top = -scale * Y_CENTER_TO_BASE - labelBgHeight
+      right = scale * X_CENTER_TO_SIDE
+      bottom = scale * Y_CENTER_TO_POINT
+      break
+    case Orientation.Left:
+      left = -scale * Y_CENTER_TO_POINT
+      top = -scale * X_CENTER_TO_SIDE
+      right = scale * Y_CENTER_TO_BASE
+      bottom = scale * X_CENTER_TO_SIDE + labelBgHeight
+      break
+    case Orientation.Right:
+      left = -scale * Y_CENTER_TO_BASE
+      top = -scale * X_CENTER_TO_SIDE
+      right = scale * Y_CENTER_TO_POINT
+      bottom = scale * X_CENTER_TO_SIDE + labelBgHeight
+      break
+  }
+  left = Math.min(left, -labelBgWidth / 2)
+  right = Math.max(right, labelBgWidth / 2)
+  const width = right - left
+  const height = bottom - top
 
   return (
-    <g className="m-vehicle-icon" transform={`scale(${scaleAmount})`}>
-      <path
-        className={"m-vehicle-icon__triangle"}
-        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
+    <svg
+      style={{ width, height }}
+      viewBox={`${left} ${top} ${width} ${height}`}
+    >
+      <VehicleIconSvg
+        size={size}
+        orientation={orientation}
+        label={label}
+        variant={variant}
       />
+    </svg>
+  )
+}
+
+export const VehicleIconSvg = ({
```

Comment:
Personally `VehicleIcon` returning an `<svg>` and `VehicleIconSvg` returning a `<g>` is not at all what I expected from the names.

Reply:
Yeah, I see that. For the caller, the important distinction is whether the component it's using should be part of an svg or whether it can be treated as an html element. Any ideas for names that would make that clear without feeling like the name contradicts the implementation?

Reply:
What about `VehicleIconSvgGroup` and then either `VehicleIconSvg` or just `VehicleIcon`? Just spitballing…

Reply:
I like `VehicleIcon` for the html version. At that point the fact that it's made with an svg can be an opaque implementation detail.

For the svg node, is the fact that it's a `<g>` important to expose? What about `...SvgNode` or `...SvgElement`?

Reply:
That makes sense. Personally I might vote for `...SvgNode` so there's no confusion about it being "an <svg> element", but whatever you prefer. Naming is hard…